runtime/near-vm-logic/src/logic.rs

```diff
@@ -278,20 +285,21 @@ impl<'a> VMLogic<'a> {
     /// * If string is longer than `max_log_len` returns `BadUtf8`.
     fn get_utf8_string(&mut self, len: u64, ptr: u64) -> Result<String> {
         let mut buf;
+        let max_len = self.config.max_log_len;
         if len != std::u64::MAX {
-            if len > self.config.max_log_len {
+            if len > max_len {
                 return Err(HostError::BadUTF8.into());
             }
             buf = Self::memory_get(self.memory, ptr, len)?;
         } else {
             buf = vec![];
-            for i in 0..=self.config.max_log_len {
+            for i in 0..=max_len {
                 Self::try_fit_mem(self.memory, ptr, i)?;
```

Comment:
need to check `i + 1` here

Reply:
similar to the comment above, we can check for `Self::try_fit_mem(self.memory, ptr + i, 1)?;`

Reply:
thanks!